interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -372,6 +407,7 @@ func (pdb *PromptDB) Prompts(user uint32) ([]*Prompt, error) {
 		// No prompts for user, but no error
 		return nil, nil
 	}
+	userEntry.expirationTimer.Reset(activityTimeout)
```

Comment:
The intention behind "activity" is whether there is an active prompt client, not whether the user is interacting with it. If there is no client, there's no way for a user to respond to prompts, so we should auto-deny them. If there is a client, we want to give the user as much time as they need to think about how they're going to respond, and let them do so. Clients shouldn't be able to run unless there's an active user session, and if there's an active user session, we never want to do something without asking the user. Worst case, the kernel or application times out the request because there's been no response. But we don't want to preemptively do that in snapd if we know there's a means for the user to be able to respond.

Reply:
Can we really conclude that the user has means to respond when we only observe queries for notices? Note, that said queries do not even need to be executed by the user themselves, but we're generally allowing any client to connect over the snapd socket, and root can make queries for any user IIRC. Maybe the initial timeout does not need to be that aggressive.

Reply:
> Can we really conclude that the user has means to respond when we only observe queries for notices?

I think yes, but I'll discuss with @sminez

Reply:
To summarize for @sminez, we've discussed a few possibilities for what to classify as "activity" sufficient to reset the prompt timeout:
1. A client requesting prompt details for any prompt assigned to the user (this is the current approach)
2. A client *replying* to any prompt assigned to the user
3. A client interacting with a dedicated prompt ACK endpoint to let snapd know that it's an active client

I still favor the former, since I think it's the most conservative about timing out prompts. If a client gets prompt details but then doesn't reply, that could be because of a problem in the client, but it could also be that the user is taking a long time to reply, and I think we want to restart the activity timeout when the prompt details were retrieved so that the user has the maximum reasonable amount of time to reply. And any future prompts that arise can re-set the timeout if the user is really taking their time.

I think a dedicated ACK endpoint would result in additional complexity and require synchronization both with the client and within the prompting backends of snapd.

If snapd doesn't timeout a prompt, worst case the application stays blocked until the kernel times out or the application times out on the syscall. In my view, the primary motivation for the timeout is that if we know there's no active client, we don't want to wait for one of those to occur. But if snapd has a hint that there is an active client, snapd should be as accommodating as it can.

Reply:
My preference would be for 1 or 3 as 2 doesn't handle what I think the most important case is, namely that a user is taking a long time to respond to a prompt while others are pending. We eagerly pull prompt details whenever the notices API gives us a new ID to process, so 1 will effectively be an ack from us that we saw the notices come through I think? So long as that is sufficient I think that's fine, but depending on how long the timeout is we may run into issues if a large number of prompts get generated at roughly the same time (though I'll admit I don't know what sort of real world scenario that would correspond to!)

Reply:
independently of how we trigger, I think we are making assumption that this are called a certain why by the API layer I fear that we need some parameters to help know that this are called actually in the context, otherwise any use of the API of this will result in being considered activity even if perhaps was something internal, or some debug code

Reply:
This was addressed in https://github.com/canonical/snapd/pull/14390/commits/4e169f78ba6a5e1a3c74fb729fea267c109f3187